### ISSUE TITLE
Diagram should describe the relationship not generate the relationship

### ISSUE THREAD

**Issue by da77a** [pgmodeler/pgmodeler#1746]
**Feature description**
I want to be able to "edit" not destroy and recreate, existing relationship definitions to change the "type" of relationship.

Assume I already have a many to one relationship by creating a foreign key constraint only.

I later realise (or change the requirements to be) that this should be 1 to 1.

All I need to do is to add a unique constraint to get the effect I want. 

If I do this "manually" the diagram still shows a "foreign key" relationship as  0:N -> 1:1 even though N *cannot* be more than 1 - because I've added a unique constraint.

If I try to create (add) a 1:1 relationship pgmodeler insists I first delete my existing foreign key, and proceeds to define a new column this is not the one I was using/want to use. Basically it turns a simple addition of a constraint into a lot of "work".

Really a relationship has *attributes* not a type - the attributes being the arity at each "End". If I have a unique constraint on the column(s) in the foreign key then the foreign key "relationship" is 0:1 not 0:N... And if I add "not null" to that it becomes 1:1.

pgmodeler could generalise the relationship types and constraint specification into one (but likely that is a big change). A more straight forward enhancement to foreign key constraint relationships only (ie as now) would just add "calculating" the "arity" by looking at the uniqueness and not null constraints and using that when "drawing" the relationship ends. 

Or maybe I'm doing something wrong and there is a way to achieve the result I want already?

**Comment by rkhaotix** [comment]
Unfortunately, it's not that simple. Relationships are the most complex part of the tool and because of the [columns/constraints propagation mechanism](https://www.pgmodeler.io/support/docs/310-relationships?v=1.0.0) pgModeler obligates the user to drop the relationship and create it again, that's because all the relationships chain is affected by a single change in one relationship. So, I think that allowing editing a relationship would cause more problems (introduce bugs) than be beneficial.

For the FK relationships, pgModeler already tries to guess the cardinality by looking into the not null and unique constraints of a table. If the relationship is not drawn correctly, maybe an adjustment would be sufficient. Anyway, I will explore this issue a bit and see what can be done.

**Comment by da77a** [comment]
> Unfortunately, it's not that simple. Relationships are the most complex part of the tool and because of the [columns/constraints propagation mechanism](https://www.pgmodeler.io/support/docs/310-relationships?v=1.0.0) 

Yes so I have read. I think the issue arises from threating these as "9 types of relationships" not "9 - or however many is natural/real" sets of relationship constraints/attributes. Noting that N:N is definitely a "special case" but the rest are imho just (some of) properties of a relationship not distinct types.

Anyway - I conceded that a change to support this full generality was not likely to be practical and obviously the existing features are used and have utility. Don't break them! 
 
> For the FK relationships, pgModeler already tries to guess the cardinality by looking into the not null and unique constraints of a table.

One of the 9 relationship types documented is *3.10.7. FK relationship*. The docs say: "This special relationship has the same semantics as the one-to-many but the user can't directly modify it".

The docs go on to describe the semantics I see in using / explicitly defining foreign keys. I do not see documented or in usage and automated/implicit change from a _one to many_ to a _one to one_ or _one to 0 or 1_ relation depending on other constraints manually added to the columns involved in the foreign key constraint and if that did happen it would make the statement *has the same semantics as the one-to-many* false.

As the *tries to guess* seems to be undocumented behaviour can you help me understand what it should do/how I should be able to get the behaviour you describe? Should it work for any foreign key relationship - including/especially where they keys / columns in the relationship are composite? Im quite happy to restrict my usage to a "foreign key and other constraint defined relationship type" if that means I can effectively edit such relationships (by changing constraints) rather than "fighting" the strongly typed and very "opinionated" (that isn't bad) diagram to generated SQL DDL relationships that are defined (and working well for what they do/define).

**Comment by rkhaotix** [comment]
Hi @da77a 

> As the tries to guess seems to be undocumented behaviour can you help me understand what it should do/how I should be able to get the behaviour you describe? Should it work for any foreign key relationship - including/especially where they keys / columns in the relationship are composite? Im quite happy to restrict my usage to a "foreign key and other constraint defined relationship type" if that means I can effectively edit such relationships (by changing constraints) rather than "fighting" the strongly typed and very "opinionated" (that isn't bad) diagram to generated SQL DDL relationships that are defined (and working well for what they do/define).

Sorry for the undocumented feature... I'm updating the docs due to the release of 1.0.0 and I'll add that missing part related to FK relationships in the docs. Anyway, I've created a small example of how pgModeler guesses the relationship type by looking into the foreign keys, unique constraints, and not null. 

So, in the image, we have the following relationships demonstrated:

1) ```(0, 1) --<>-- (0, N)``` between ```table_a``` and ```table_b```
2) ```(1, 1) --<>-- (0, N)``` between ```table_a``` and ```table_c```
3) ```(1, 1) --<>-- (0, 1)``` between ```table_a``` and ```table_d```
4) ```(0, 1) --<>-- (0, 1)``` between ```table_a``` and ```table_e```

![fk_rels_samples](https://user-images.githubusercontent.com/2205476/220119220-537f06e9-6018-42f3-9c44-9515b59d2da2.png)

[fk_rels_samples.zip](https://github.com/pgmodeler/pgmodeler/files/10784713/fk_rels_samples.zip)
